firefox-ios/Client/Application/AppDelegate.swift

```diff
@@ -268,6 +269,46 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Since we only need the topSites data in the archiver, let's write it
         widgetManager?.writeWidgetKitTopSites()
     }
+
+    // MARK: Telemetry
+    private func sendStartupTelemetry() {
+        _ = profile.autofill.reopenIfClosed()
+        profile.autofill.listCreditCards(completion: { cards, error in
+            guard let cards = cards, error == nil else { return }
+            self.sendCreditCardsSavedAllTelemetry(numberOfSavedCreditCards: cards.count)
+        })
+
+        let searchController = UISearchController()
+        let loginsViewModel = PasswordManagerViewModel(
+            profile: profile,
+            searchController: searchController,
+            theme: LightTheme()
+        )
+        let dataSource = LoginDataSource(viewModel: loginsViewModel)
+        loginsViewModel.loadLogins(loginDataSource: dataSource)
+
+        loginsViewModel.queryLogins("") { logins in
+            self.sendLoginsSavedAllTelemetry(numberOfSavedLogins: logins.count)
+        }
+    }
+
+    private func sendLoginsSavedAllTelemetry(numberOfSavedLogins: Int) {
+        let savedLoginsExtra = [TelemetryWrapper.EventExtraKey.loginsQuantity.rawValue: Int64(numberOfSavedLogins)]
+        TelemetryWrapper.recordEvent(category: .information,
+                                     method: .foreground,
+                                     object: .loginsSavedAll,
+                                     extras: savedLoginsExtra)
+    }
+
+    private func sendCreditCardsSavedAllTelemetry(numberOfSavedCreditCards: Int) {
```

Comment:
@petruSt24 Few questions 
a) why do we want to send num of cc saved on each launch? 
b) also, why can't we do it as part of when browser view controller loads and why does it need to happen in AppDelegate?
c) If you are sending it on every launch then did you ask if we are to fully remove it from the cc page?

Lastly I don't feel comfortable bloating AppDelegate with these types of changes, can we not add a separate class or struct for a cleaner approach if we do have to do these changes?